tensorflow/contrib/tensorrt/convert/convert_graph.cc

```diff
@@ -357,7 +357,18 @@ tensorflow::Status GetEngineInfo(
     auto node = *it;
     auto node_device = node->requested_device();
     if (!node_device.empty()) {
-      segment_devices.insert(node_device);
+      // If device is CPU, we should've caught that in the segmenter. Don't add
+      // CPU to segment_device because that would cause a segfault in
+      // GetDeviceAndAllocator.
+      DeviceNameUtils::ParsedName parsed_name;
+      DeviceNameUtils::ParseFullName(node_device, &parsed_name);
+      if (parsed_name.type == "CPU") {
```

Comment:
Can we add a test for this fix as well?

Reply:
This fix is covered by int32_test. This change is more of a final safeguard to prevent a segfault in case the validator didn't exclude an INT32 node. Since the validator for matmul and binary op prevent the nodes in int32_test from being converting, this line isn't reached. But if you removed the type check in the validator, this code would be executed.